### PR TITLE
first pass, harvester cloud provider provisioning automation p0 use case

### PR DESCRIPTION
**Pull request by slickwarren**
This is a first PR for me using the GO framework; referenced https://github.com/rancher/rancher/pull/35829 for much of the changes (also had to pull some of that code which might cause conflicts later, since that one isn't merged)

Here's the basic net-new from Israel's PR:
* harvester cloud credentials
* new test that uses harvester cloud creds + the provisioning cases outlined in Israel's PR.  
* some updates to the way a standard user is created (random string attached to name, still could be improved)

Tests:
* successful run was completed using the attached `test.go` file in my local harvester setup

next up: make each of the hard-coded values a variable for the user to enter.

### CODE REVIEW

File: tests/v2/provisioining/provisioining_harvester_node_driver_test.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+		})
+	}
+}
+
+// func (r *RKE2NodeDriverProvisioningTestSuite) TestProvisioning_RKE2HarvesterClusterDynamicInput() {
+// 	nodesAndRoles := NodesAndRolesInput()
+// 	if len(nodesAndRoles) == 0 {
+// 		r.T().Skip()
+// 	}
+
```
Comment (igomez06) [inline tests/v2/provisioining/provisioining_harvester_node_driver_test.go:162]: Did you test out with the node inputs? Once we get my PR in, for sure want to have this working. Also I think these tests can go in provisioning_node_driver_test.go, once we get my PR merged in.

**Review by igomez06 — COMMENTED** [review]
Nice work, this is it. Once we get my PR merged, we can rework this a little. There's more detail in another comment I left. Other than that LGTM good work.

File: tests/framework/extensions/machinepools/harvester_machine_config.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// It returns an *unstructured.Unstructured that CreateMachineConfig uses to created the rke-machine-config
+func NewHarvesterMachineConfig(generatedPoolName, namespace, network, image string) *unstructured.Unstructured {
+	machineConfig := &unstructured.Unstructured{}
+	machineConfig.SetAPIVersion("rke-machine-config.cattle.io/v1")
+	machineConfig.SetKind(HarvesterKind)
+	machineConfig.SetGenerateName(generatedPoolName)
+	machineConfig.SetNamespace("fleet-default")
+
+	machineConfig.Object["diskSize"] = "40"
+	machineConfig.Object["cpuCount"] = "2"
```
Comment (igomez06) [inline tests/framework/extensions/machinepools/harvester_machine_config.go:21]: I think we'll probably have these in a config file. But my PR will have those details and you build off of that work

File: tests/v2/provisioining/provisioining_harvester_node_driver_test.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+			clusterName := GenerateRandomName(baseHarvesterClusterName)
+			generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+			machinePoolConfig := machinepools.NewHarvesterMachineConfig(generatedPoolName, "default", "default/ctw-network-1", "default/image-rpj98")
+
+			machineConfigResp, err := machinepools.CreateMachineConfig(machinepools.HarvesterResourceConfig, machinePoolConfig, testSessionClient)
+			require.NoError(r.T(), err)
+
+			machinePools := machinepools.RKEMachinePoolSetup(tt.nodeRoles, machineConfigResp)
+
+			cluster := clusters.NewRKE2ClusterConfig(clusterName, "fleet-default", "calico", cloudCredential.ID, "v1.21.7+rke2r2", machinePools)
```
Comment (sowmyav27) [inline tests/v2/provisioining/provisioining_harvester_node_driver_test.go:136]: Can we parameterize the RKE2 version, network provider - Calico here ?

Comment (sowmyav27) [inline tests/v2/provisioining/provisioining_harvester_node_driver_test.go:136]: Can we parameterize the RKE2 version, network provider - Calico here ?

File: tests/v2/provisioining/provisioining_harvester_node_driver_test.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+		r.Run(tt.name, func() {
+			testSession := session.NewSession(r.T())
+			defer testSession.Cleanup()
+
+			testSessionClient, err := tt.client.WithSession(testSession)
+			require.NoError(r.T(), err)
+
+			clusterName := GenerateRandomName(baseHarvesterClusterName)
+			generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+			machinePoolConfig := machinepools.NewHarvesterMachineConfig(generatedPoolName, "default", "default/ctw-network-1", "default/image-rpj98")
```
Comment (sowmyav27) [inline tests/v2/provisioining/provisioining_harvester_node_driver_test.go:129]: Can we parameterize these too. ?

**Review by sowmyav27 — CHANGES_REQUESTED** [review]

File: tests/v2/provisioining/provisioining_harvester_node_driver_test.go (hunk 4 above)
Comment (igomez06) [reply tests/v2/provisioining/provisioining_harvester_node_driver_test.go:129]: These could be a part of the config file.

File: tests/v2/provisioining/provisioining_harvester_node_driver_test.go (hunk 3 above)
Comment (igomez06) [reply tests/v2/provisioining/provisioining_harvester_node_driver_test.go:136]: These could be a part of the config file.

**Pull request closed without merging**